### PR TITLE
Readability improvements

### PR DESCRIPTION
**Pull request by robertbastian**
This started out as docs improvements, but I've also touched non-docs code now.

```rust
let collator: Collator = Collator::try_new(...);
```
becomes
```rust
let collator = Collator::try_new(...);
```
======
```rust
let asdf = foo.iter().map(...).collect::<Vec<_>>();
```
becomes
```rust
let asdf = Vec::from_iter(foo.iter().map(...));
```
(this shorter, less symbol salad, does `into_iter()`, and puts the type upfront)

======

```rust
let asdf: ZeroMap<u8, u8> = ZeroMap::from_foo();
```
becomes
```rust
let asdf = ZeroMap::<u8, u8>::from_foo();
```
(this is shorter, and better for refactorings, as the expression has a type)

======

```rust
let loc: Locale = "foo".parse().unwrap();
```
becomes
```rust
let loc = "foo".parse::<Locale>().unwrap();
```
(this is better for refactorings, as the expression has a type)

### CODE REVIEW

**Comment by sffc** [comment]
I don't think I like changing `foo.iter().map(...).collect()` because it doesn't chain, which negatively impacts indentation. The rest seem fine.

**Review by sffc — CHANGES_REQUESTED** [review]
Also, CI is failing.

**Comment by sffc** [comment]
It seems self-inconsistent that you're changing code to use turbofish in `.parse::<Foo>()` but not `.collect::<Foo>()`

File: components/calendar/src/provider.rs [hunk 1]
```diff
@@ -273,9 +273,11 @@ impl serde::Serialize for WeekdaySet {
...
     {
         if serializer.is_human_readable() {
-            crate::week_of::WeekdaySetIterator::new(IsoWeekday::Monday, *self)
-                .collect::<alloc::vec::Vec<_>>()
-                .serialize(serializer)
+            alloc::vec::Vec::from_iter(crate::week_of::WeekdaySetIterator::new(
+                IsoWeekday::Monday,
+                *self,
+            ))
+            .serialize(serializer)
```
Comment (sffc) [inline components/calendar/src/provider.rs:280]: Something like this. I find the old code more readable.

```rust
// Old
            crate::week_of::WeekdaySetIterator::new(IsoWeekday::Monday, *self)
                .collect::<alloc::vec::Vec<_>>()
                .serialize(serializer)

// New
            alloc::vec::Vec::from_iter(crate::week_of::WeekdaySetIterator::new(
                IsoWeekday::Monday,
                *self,
            ))
            .serialize(serializer)
```

File: components/calendar/src/week_of.rs [hunk 2]
```diff
@@ -714,34 +714,38 @@ fn test_weekend() {
...
     assert_eq!(
-        WeekCalculator::try_new(&locale!("und").into())
-            .unwrap()
-            .weekend()
-            .collect::<Vec<_>>(),
+        Vec::from_iter(
+            WeekCalculator::try_new(&locale!("und").into())
+                .unwrap()
+                .weekend()
+        ),
```
Comment (sffc) [inline components/calendar/src/week_of.rs:721]: Here's a case where we need to increase indentation.

```rust
// Max indent 3
        WeekCalculator::try_new(&locale!("und").into())
            .unwrap()
            .weekend()
            .collect::<Vec<_>>(),

// Max indent 4
        Vec::from_iter(
            WeekCalculator::try_new(&locale!("und").into())
                .unwrap()
                .weekend()
        ),
```

**Comment by sffc** [comment]
I'm being somewhat picky with this review because the PR is literally entitled "Readability improvements" and I disagree that these are all consistently improvements.

**Comment by robertbastian** [comment]
> It seems self-inconsistent that you're changing code to use turbofish in .parse::<Foo>() but not .collect::<Foo>()

I would love to change code to `Foo::from_str`, but that's not in the prelude, whereas `FromIterator` is. I think it's more important to have expressions that don't rely on surrounding type context than to avoid turbofishes, because things like `let x: Foo = "foo".parse().unwrap();` are a pain during refactorings.

**Comment by sffc** [comment]
> > It seems self-inconsistent that you're changing code to use turbofish in .parse::() but not .collect::()
> 
> I would love to change code to `Foo::from_str`, but that's not in the prelude, whereas `FromIterator` is. I think it's more important to have expressions that don't rely on surrounding type context than to avoid turbofishes, because things like `let x: Foo = "foo".parse().unwrap();` are a pain during refactorings.

So it sounds like you would be in support of `let x = y.iter().collect::<Foo>()`, right? Just not `let x: Foo = y.iter().collect()`? I'm onboard with that; type inference can cause problems. I just feel that `.collect()` is more conventional in Rust.

Evidence: [98.8k instances of `from_iter(`](https://github.com/search?q=from_iter%28+language%3ARust+&type=code) versus [1.3M instances of `collect(`](https://github.com/search?q=collect%28+language%3ARust+&type=code) on Rust GitHub.

**Review by sffc — COMMENTED** [review]
It seems we're not in alignment on what is more readable, so I added "discuss-priority"

**Pull request closed without merging**